### PR TITLE
Benchmarks

### PR DESCRIPTION
**Pull request by fsaintjacques**
This adds a new builder type and docker image.

### CODE REVIEW

File: ursabot/docker.py [hunk 1]
```diff
@@ -275,6 +275,8 @@ def conda(*packages, files=tuple()):
     'libboost-system-dev',
     'python',
     'python-pip',
+    'python3',
```
Comment (kszucs) [inline ursabot/docker.py]: We'll need a nice way to handle python versions uniformly.

File: ursabot/docker.py [hunk 2]
```diff
@@ -368,6 +370,18 @@ def conda(*packages, files=tuple()):
...
+for arch in ['amd64']:
+    os = 'ubuntu-18.04'
+    base = f'{arch}/ubuntu:18.04'
+    steps = [RUN(apt(*ubuntu_pkgs + ['libbenchmark-dev']))]
+
+    cpp = DockerImage('cpp', base=base, arch=arch, os=os, variant='benchmark',
+                      steps=steps)
+    cpp_worker = DockerImage('cpp', base=cpp, tag='worker', steps=worker_steps)
+
+    arrow_images.extend([cpp, python, cpp_worker, python_worker])
```
Comment (kszucs) [inline ursabot/docker.py]: Only add cpp and cpp_worker

File: ursabot/builders.py [hunk 3]
```diff
@@ -285,6 +293,20 @@ class ArrowCppTest(Builder):
     ]
 
 
+class ArrowCppBenchmark(Builder):
+    tags = ['arrow', 'cpp']
```
Comment (kszucs) [inline ursabot/builders.py]: Benchmark perhaps?

File: ursabot/docker.py (hunk 1 above)
Comment (fsaintjacques) [reply ursabot/docker.py]: yeah, I didn't want to depend on the conda version.

File: ursabot/docker.py (hunk 2 above)
Comment (fsaintjacques) [reply ursabot/docker.py]: derp.

File: ursabot/builders.py (hunk 3 above)
Comment (fsaintjacques) [reply ursabot/builders.py]: Already implicit via the docker image (I added it here first, but then it complained about duplicate tags).

**Pull request merged**